OpenSkyPlusApi/OpenSkyPlusApi.cs

```diff
@@ -1,722 +1,754 @@
-
-    public static void Initialize()
-    {
-        OpenSkyPlusApiInjector.MessageDeviceReady += log =>
-        {
-            try
-            {
-                OpenSkyPlusApi.GetInstance(log);
-                MessageApiLoaded?.Invoke();
-            }
-            catch (OpenSkyPlusApiException) { }
-            catch (Exception ex)
-            {
-                OpenSkyPlus.Log.LogError($"Failed to create an Api singleton instance: {ex}");
-            }
-        };
-    }
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using BepInEx.Logging;
+using OpenSkyPlusApi;
+
+namespace OpenSkyPlus;
+
+public delegate void NotificationApiLoaded();
+
+public delegate void NotificationDeviceReady(ManualLogSource log);
+
+public delegate void NotificationMonitorStatusChange();
+
+public class OpenSkyPlusApi : AbstractOpenSkyPlusApi
+{
+    public static readonly string SupportedVersion = "4.4.7";
+    private static ManualLogSource _logger;
+    private static int _batteryLevel;
+    private static Handedness _handedness = Handedness.Right;
+
+    private static readonly Lazy<OpenSkyPlusApi> Instance = new(() => new OpenSkyPlusApi());
+    private readonly OpenSkyPlusConfiguration.Configuration _config = OpenSkyPlusConfiguration.Config;
+
+    private OpenSkyPlusApi()
+    {
+        DeviceControls.Initialize(_logger);
+        DeviceControls.CheckAssemblyAndLicense();
+        OpenSkyPlusApiInjector.MessageNewShot += ReceiveShot;
+        OpenSkyPlusApiInjector.MessageMonitorConnected += Connected_;
+        OpenSkyPlusApiInjector.MessageMonitorDisconnected += Disconnected;
+        DeviceControls.MessageMonitorReady += Ready_;
+        DeviceControls.MessageMonitorNotReady += NotReady;
+
+        _logger.LogDebug($"{_config.AppSettings.LaunchMonitor} API is initialized");
+    }
+
+    private static bool Connected { get; set; }
+
+    private static bool Ready { get; set; }
+
+    private static int BatteryLevel
+    {
+        get => _batteryLevel;
+        set => _batteryLevel = value > 100 ? 100 : value;
+    }
+
+    public bool Charging { get; set; } = false;
+
+    private ShotMode ShotMode { get; set; } = ShotMode.Normal;
+
+    private ShotData LastShot { get; set; } = new();
+
+    protected override void Initialize()
+    {
+    }
+
+    public static OpenSkyPlusApi GetInstance(ManualLogSource log = null)
+    {
+        if (log != null && !Instance.IsValueCreated) _logger = log;
+        return Instance.Value;
+    }
+
+    public static string GetApiVersion()
+    {
+        return ApiVersion;
+    }
+
+    public static bool IsLoaded()
+    {
+        return Instance.IsValueCreated;
+    }
+
+    public override bool IsConnected()
+    {
+        return Connected;
+    }
+
+    public override bool IsReady()
+    {
+        return Ready;
+    }
+
+    public override bool ReadyForNextShot()
+    {
+        try
+        {
+            if (DeviceControls.Armed == true)
```

Comment:
this is getting defaulted to true quit a bit. cleaner to just set Armed property to default to true whenever its fetched

Reply:
I could not find a way to get the status from the box.  That would be the best but I did not find the entrypoint/property/method to perform the task yet.